### ISSUE TITLE
Revamp event management

### ISSUE THREAD

**Issue by ged-odoo** [odoo/owl#111]
# Motivation

There are currently two event systems in parallel:
- dom events: regular events triggered on dom nodes...
- "business" events: event triggered on `EventBus` instances (so, on widgets)

To register handlers for these events, we can do `t-on` on dom nodes (which then will listen on dom events)  or on `t-widget` (which will listen on "business" events).

As a consequence, it is not possible to listen on click events on a component.  Also, those business events do not bubble up.

# Spec

In short: remove "business" event from component, they should trigger dom events on their root node
